src/dagmc.cpp

```diff
@@ -20,28 +38,100 @@ moab::DagMC* DAG;
 
 } // namespace model
 
+
+bool get_uwuw_materials_xml(std::string& s) {
+  UWUW uwuw(DAGMC_FILENAME.c_str());
+
+  std::stringstream ss;
+  bool uwuw_mats_present = false;
+  if (uwuw.material_library.size() != 0) {
+    uwuw_mats_present = true;
+    // write header
+    ss << "<?xml version=\"1.0\"?>\n";
+    ss << "<materials>\n";
+    std::map<std::string, pyne::Material> mat_lib = uwuw.material_library;
+    // write materials
+    for (auto mat : mat_lib) { ss << mat.second.openmc("atom"); }
+    // write footer
+    ss << "</materials>";
+    s = ss.str();
+  }
+
+  return uwuw_mats_present;
+}
+
+pugi::xml_document* read_uwuw_materials() {
+  pugi::xml_document* doc = NULL;
+
+  std::string s;
+  bool found_uwuw_mats = get_uwuw_materials_xml(s);
+  if (found_uwuw_mats) {
+    doc = new pugi::xml_document();
+    pugi::xml_parse_result result = doc->load_string(s.c_str());
+  }
+  return doc;
+}
+
+bool write_uwuw_materials_xml() {
```

Comment:
This function doesn't look like it's used anywhere?

Reply:
True. I'm thinking it might come in handy later as a way to generate a `materials.xml` file via the CAPI. Maybe I should just go ahead include that here then...

Reply:
That's ok -- if there's an intended use later we can keep it in. No need to add the `materials.xml` generation now unless you feel strongly